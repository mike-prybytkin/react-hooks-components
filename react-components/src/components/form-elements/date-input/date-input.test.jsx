import React from 'react';
import { render } from '@testing-library/react';
import mockText from '../../../mocks/text';
import DateInput from './date-input';

const setUp = () => {
  return render(
    <DateInput
      labelType={mockText.labelUserBirthday}
      defaultValue="2000-01-01"
      inputDateRef={undefined}
      id={'test'}
    />
  );
};

describe('DateInput component', () => {
  it('should correctly render date input with a className', () => {
    const { container } = setUp();
    expect(container.getElementsByClassName('date-input').length).toBe(1);
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
