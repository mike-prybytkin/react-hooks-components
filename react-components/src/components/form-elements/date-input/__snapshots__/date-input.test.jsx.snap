// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="date-input"
  >
    <label
      class="date-input__label"
    >
      Birthday:
      <input
        class="date-input__input"
        id="test"
        type="date"
        value="2000-01-01"
      />
    </label>
  </div>
</DocumentFragment>
`;
